src/FormBuilder/resources/js/form-builder/src/components/canvas/FormSettings.tsx

```diff
@@ -0,0 +1,91 @@
+import React, {useMemo} from 'react';
+import FormSettingsContainer from '@givewp/form-builder/components/canvas/FormSettingsContainer';
+import FormGeneralSettingsGroup from '@givewp/form-builder/settings/group-general';
+import FormDonationConfirmationSettingsGroup from '@givewp/form-builder/settings/group-donation-confirmation';
+import {__} from '@wordpress/i18n';
+import EmailGeneralSettings from '@givewp/form-builder/settings/group-email-settings/general';
+import getEmailSettings from '@givewp/form-builder/settings/group-email-settings';
+
+/**
+ * @unreleased
+ */
+export default function FormSettings() {
+    const routes: Route[] = useMemo(() => {
+        const additionalSettings = validateRoutes(
+            wp.hooks.applyFilters('givewp_form_builder_settings_additional_routes', [])
+        );
+
+        return [
+            {
+                name: __('General', 'give'),
+                path: 'general',
+                element: FormGeneralSettingsGroup,
+            },
+            {
+                name: __('Donation Confirmation', 'give'),
+                path: 'donation-confirmation',
+                element: FormDonationConfirmationSettingsGroup,
+            },
+            {
+                name: __('Email Settings', 'give'),
+                path: 'email-settings',
+                element: null,
+                childRoutes: [
+                    {
+                        name: __('General', 'give'),
+                        path: 'email-settings/general',
+                        element: EmailGeneralSettings,
+                    },
+                    ...getEmailSettings(),
+                ],
+            },
+            ...additionalSettings,
+        ];
+    }, []);
+
+    return <FormSettingsContainer routes={routes} />;
+}
+
+/**
+ * @unreleased
+ */
+function validateRoutes(routes: Route[]): Route[] {
+    return routes.filter((route) => {
+        let isValid = isValidRoute(route);
```

Comment:
Let's also validate the path name to make sure there are no conflicts.  We can use `validateRoutes()` for the entire routes array as previously mentioned.  We should also do some kind of error logging - nothing fatal but at least a console error to inform the implementer.

Reply:
Resolved: e44f5683ac19baf68dbe9e8d9f044ae976a49eac